src/engine/renderer/tr_shade.cpp

```diff
@@ -1977,14 +1989,22 @@ static void Render_forwardLighting_DBS_directional( shaderStage_t *diffuseStage,
 	// bind u_NormalFormat
 	gl_forwardLightingShader_directionalSun->SetUniform_NormalFormat( tess.surfaceShader->normalFormat );
 
-	// bind u_SpecularMap
-	if ( r_forceSpecular->integer )
-	{
-		GL_BindToTMU( 2, diffuseStage->bundle[ TB_DIFFUSEMAP ].image[ 0 ] );
-	}
-	else if ( diffuseStage->bundle[ TB_SPECULARMAP ].image[ 0 ] )
+	if ( specularMapping )
```

Comment:
This changes the effect of `r_forceSpecular`. Before, it would do something even without a specular map image, but now it doesn't.

Reply:
hmmm, what is “_`r_forceSpecular` without specular map image_” for?

Reply:
I don't know, probably none will shed a tear if you delete it.

Reply:
deleted